### ISSUE TITLE
Streaming error mishandled in http.rb

### ISSUE THREAD

**Issue by strukturedkaos** [alexrudall/anthropic#20]
[Line 154 in http.rb](https://github.com/alexrudall/anthropic/blob/b9a3a6f84bdbba018660df5b186bbb6abc309a9f/lib/anthropic/http.rb#L154) on `env` is `nil` on valid requests. This causes `merge` to error on [line 157](https://github.com/alexrudall/anthropic/blob/b9a3a6f84bdbba018660df5b186bbb6abc309a9f/lib/anthropic/http.rb#L157)

```ruby
 return unless env&.status != 200
```

Example:

```ruby
          response = anthropic_client.messages(
            parameters: {
              temperature: 0.7,
              model: model,
              messages: message_data
              max_tokens: 100,
              preprocess_stream: :text,
              stream: proc do |incremental_response, delta|
                next if delta.blank?
                Rails.logger.info "Incremental response: #{incremental_response}"
                Rails.logger.info "Delta: #{delta}"
              end
            })  
```

**Comment by alexrudall** [comment]
Thanks, @strukturedkaos. Have you seen this occur @swombat ?

**Comment by redders123** [comment]
Any chance of this being fixed in the next few days? We can't use the streaming api right now.

**Comment by leo-quimbee** [comment]
Also getting this error

**Comment by alexrudall** [comment]
@redders123 will see if I can fix this

**Comment by strukturedkaos** [comment]
@alexrudall Anything I can help with? Are you able to reproduce the issue?

In case it helps:

ruby 3.2.2 
Rails 6.1.7.3

**Comment by alexrudall** [comment]
Hey @strukturedkaos -- struggling to find the time to look at this - but I'm very open to a PR if you're able to figure out a fix! If not I will get to it when I can.

**Comment by geeosh** [comment]
I ran into this as well, it's because the streaming proc method signature changed in Faraday 2.5.0, and any older version only provides the first two arguments to the proc ([see 6799f58](https://github.com/lostisland/faraday/commit/6799f5852d31dae32699949790633c68282ab71d)).

So you could update the gemspec to require ~> 2.5.0 of Faraday, although that wouldn't work for me as I have other gems that require < 2.0 of Faraday that I'm not able to easily upgrade.

The other option is to update `Anthropic::HTTP#handle_faraday_error` to accomodate the case where `env` isn't provided as a third argument. At first glance I don't see any way to get the http request status in order to check if there is an error with an older version of Faraday, so you might just have to assume the streaming request is successful on older versions of Faraday? @alexrudall if you have any suggestions about how you'd like to handle older versions of Faraday and I can put together a PR.

**Comment by strukturedkaos** [comment]
@geeosh your fix worked for me. Thank you. Hopefully we can get the PR merged soon.

**Comment by alexrudall** [comment]
Thanks for your patience @geeosh & @strukturedkaos, @leo-quimbee, @redders123, now released in [v0.3.2](https://rubygems.org/gems/anthropic)
